docs/DeveloperGuide.adoc

```diff
@@ -137,256 +167,944 @@ image:BetterModelClassDiagram.png[]
+image::add-transaction/AddTransaction0.png[]
 
-Given below is an example usage scenario and how the undo/redo mechanism behaves at each step.
+Step 2. The user executes **add_expense n/Laksa v/3.50** to insert an `**Expense**` with `**Description**` set to "Laksa"
+and `**Value**` set to "3.50". The `**TransactionDate**` is set to the user's current system date in the form "dd/mm/yyyy".
+The input is now checked and an attempt to parse each parameter occurs:
 
-Step 1. The user launches the application for the first time. The `VersionedAddressBook` will be initialized with the initial address book state, and the `currentStatePointer` pointing to that single address book state.
+* `**Description**` is parsed by `AddTransactionCommandParser#parseTransactionDescription(ArgumentMultimap)`
+* `**Value**` is parsed by `AddTransactionCommandParser#parseTransactionValue(ArgumentMultimap)`
+* `**TransactionDate**` is instantiated by `AddTransactionCommandParser#parseTransactionDate()`
 
-image::UndoRedoState0.png[]
+TIP: `**ArgumentMultimap**` is a class that stores all the parsed parameters taken from the user input.
 
-Step 2. The user executes `delete 5` command to delete the 5th person in the address book. The `delete` command calls `Model#commitAddressBook()`, causing the modified state of the address book after the `delete 5` command executes to be saved in the `addressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted address book state.
+Since the user input is valid, the `**Expense**` is successfully created and inserted into the transaction list.
+The transaction list now contains 1 `**Transaction**` object.
 
-image::UndoRedoState1.png[]
+image::add-transaction/AddTransaction1.png[]
 
-Step 3. The user executes `add n/David ...` to add a new person. The `add` command also calls `Model#commitAddressBook()`, causing another modified address book state to be saved into the `addressBookStateList`.
+Step 3. The user executes **add_income n/Bursary v/500 r/Awarded for doing well in school** to insert an `**Income**` with
+the `**Description**` set to "Bursary", `**Value**` set to "500" and `**Remark**` set to "Awarded for doing well in school".
+The `**TransactionDate**` is set to the user's current system date in the form "dd/mm/yyyy". The input is now checked in
+a similar fashion as in Step 2 except for:
 
-image::UndoRedoState2.png[]
+* `**Remark**` is parsed by `AddTransactionCommandParser#parseTransactionRemark(ArgumentMultimap)`
+
+Again, since the input is valid, the `**Income**` is successfully added into the transaction list. The transaction list
+now contains 2 `**Transaction**` objects.
+
+image::add-transaction/AddTransaction2.png[]
+
+The following activity diagram summarizes what happens when the user executes a command to add a new `**Transaction**`:
+
+.Activity diagram of adding a transaction into the transaction list
+image::add-transaction/AddTransactionSequence.png[]
```

Comment:
You _may_ want to consider removing the hexagons? I'm not sure if they're allowed.